### PR TITLE
Standardize on `metabase.util.log` for logging; use it everywhere

### PR DESCRIPTION
**Pull request by bshepherdson**
This has the same interface as `clojure.tools.logging` but supports CLJS
as well. It was formerly known as `metabase.shared.util.log`.

### CODE REVIEW

File: src/metabase/util/log.clj [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+  `(logp :error ~@args))
+
+(defmacro errorf
+  "Log a message at the `:error` level by applying `format` to a format string and args."
+  {:arglists '([format-string & args] [e format-string & args])}
+  [& args]
+  `(logf :error ~@args))
+
+;; TODO Uncomment this when it's needed.
+#_(defmacro fatal
```
Comment (metamben) [inline src/metabase/util/log.clj:149]: Why is this commented out?

File: src/metabase/util/log.clj [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+                         (with-out-str
+                           (pprint/with-pprint-dispatch pprint/code-dispatch
+                             (pprint/pprint '~expr)
+                             #_{:clj-kondo/ignore [:discouraged-var]}
+                             (print "=> ")
+                             (pprint/pprint %)))))
+     :clj  `(clojure.tools.logging/spy ~level ~expr))))
+
+;; TODO Uncomment this when it's needed.
+#_(defmacro spyf
```
Comment (metamben) [inline src/metabase/util/log.clj:179]: Why is this commented out?

**Review by metamben — APPROVED** [review]
Nice!

File: src/metabase/util/log.clj (hunk 1 above)
Comment (bshepherdson) [reply src/metabase/util/log.clj:149]: There are no users of it anywhere. I'd rather not delete it, hence it's commented out and has a TODO. Should I rephrase the TODO to make it more clear?

File: src/metabase/util/log.clj [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+           (clojure.tools.logging/log* logger# ~level x#  ~(if (nil? more)
+                                                             ""
+                                                             `(print-str ~@more)))
+           (clojure.tools.logging/log* logger# ~level nil (print-str x# ~@more)))))))
+
+(defn- tools-logf
+  "Macro helper for [[logf]] in CLJ."
+  [logger-ns level x more]
+  (if (or (instance? String x) (nil? more))
+    ;; Simple case: just a String and no args.
```
[the remark this answers is not in the document]
Comment (bshepherdson) [reply src/metabase/util/log.clj:57]: I think I was describing the two simple cases that are handled the same way. I've clarified the comment.

File: src/metabase/util/log.clj (hunk 1 above)
Comment (bshepherdson) [reply src/metabase/util/log.clj:149]: I'll uncomment this and ignore the Kondo warning.

**Pull request merged**